charts/kube-master/templates/controller-manager.yaml

```diff
@@ -77,24 +79,38 @@ spec:
       containers:
         - name: controller-manager
           image: {{ include "hyperkube.image" . | quote }}
+          command:
+            - /bin/sh
           args:
-            - /hyperkube
-            - controller-manager
-            - --allocate-node-cidrs=true
-            - --cloud-config=/etc/kubernetes/cloudprovider/openstack.config
-            - --cloud-provider=openstack
-            - --cluster-cidr={{ .Values.clusterCIDR }}
-            - --cluster-name=kubernetes
-            - --cluster-signing-cert-file=/etc/kubernetes/certs/apiserver-nodes-ca.pem
-            - --cluster-signing-key-file=/etc/kubernetes/certs/apiserver-nodes-ca-key.pem
-            - --configure-cloud-routes=true
-            - --controllers=*,bootstrapsigner,tokencleaner
-            - --kubeconfig=/etc/kubernetes/config/kubeconfig
-            - --leader-elect=false
-            - --root-ca-file=/etc/kubernetes/certs/tls-ca.pem
-            - --service-account-private-key-file=/etc/kubernetes/certs/apiserver-clients-ca-key.pem
-            - --service-cluster-ip-range={{ .Values.serviceCIDR }}
-            - --use-service-account-credentials
+            - -ce
+            - |-
+              while ! wc /tmp/reject-cert/ca.crt; do sleep 1; done
```

Comment:
I assume this is required because the cert gets generated by the sidecar. I would suggest to have an init container using the same image than the sidecar. This sidecar should generate the tickets during pod initialization (e.g. ` /reject-proxy -cert-dir=/tmp/reject-cert generate-ca`)
Using that its always clear that the cert has been generated when both containers (cm and sidecar) are starting.

Reply:
I changed this everywhere.